openwisp_controller/config/admin.py

```diff
@@ -534,14 +535,28 @@ def get_inlines(self, request, obj):
                 inlines.append(inline)
         return inlines
 
-    def reversion_register(self, model, **options):
-        if model == Device:
-            options['follow'] = (
-                *(options['follow']),
-                'deviceconnection_set',
-                'devicelocation',
-            )
-        return super().reversion_register(model, **options)
+    @classmethod
+    def add_reversion_following(cls, follow):
+        # DeviceAdmin is used by other modules that register InlineModelAdmin
+        # using monkey patching. The default implementation of reversion.register
+        # ignore such inlines and does not update the "follow" field accordingly.
+        # This method updates the "follow" fields of the Device model
+        # by unregistering the Device model from reversion and re-registering it.
+        # Only the "follow" option is updated.
+        device_reversion_options = reversion.revisions._registered_models[
+            reversion.revisions._get_registration_key(Device)
+        ]
+        following = set(device_reversion_options.follow) | set(follow)
```

Comment:
Is this a union right? Can it be rewritten using the union method of set please? This will make the code more readable since the bitwise operator is almost not used at all in OpenWISP.